kitsune/sumo/forms.py

```diff
@@ -20,32 +17,22 @@ def __init__(self, *args, **kwargs):
     def clean(self, *args, **kwargs):
         """Generic clean method used by all forms in the question app.
 
-        Parse content for any toll free numbers.
+        Parse content for suspicious content.
+        - Toll free numbers
+        - Toll vanity numbers
+        - Links - not necessarily spam content
         """
+
         cdata = self.cleaned_data.get('content')
         if not cdata:
             return super(KitsuneBaseForumForm, self).clean(*args, **kwargs)
 
-        # At the moment validation doesn't reach this point b/c the logout and deactivation
-        # is_spam = is_toll_free_number(cdata)
-
-        # if is_spam:
-        #     self.cleaned_data.update({
-        #         'is_spam': True
-        #     })
-
         if not self.user:
             raise forms.ValidationError('Something went terribly wrong. Please try again')
 
-        # Allow moderators to post anything
-        has_links = False
-        if not self.user.has_perm('flagit.can_moderate'):
-            if switch_is_active('disallow_all_links'):
-                try:
-                    has_links = match_regex_with_timeout(settings.SIMPLE_DOMAIN_IP_REGEX, cdata)
-                except TimeoutError:
-                    has_links = True
-
-        if has_links:
-            raise forms.ValidationError('Links are not allowed in the forums.')
+        if not self.user.has_perm('flagit.can_moderate') and check_for_spam_content(cdata):
```

Comment:
Will we add the exclusion for trusted contributors in a later commit?

Reply:
added